src/nexus/utils.py

```diff
@@ -28,7 +28,11 @@ def check_availability(self) -> bool:
         response = self.http_session.get(self.api_base_url, timeout=self.request_timeout_seconds)
         return response.status_code == requests.codes.ok
 
-    def get_total_records(self, endpoint: str) -> int:
+    def get_total_records(
+            self,
+            endpoint: str,
+            params: dict = dict
```

Comment:
Using `dict` as a type annotation is not supported on Python < 3.9, we should use `typing.Dict` instead, since AFAIK GeoNode wants to support at least Python 3.8. 

Moreover, passing `dict` as the default value for params seems like a subtle bug. `dict` is a type that, when called, creates dictionaries. In other words:

```python
dict == {}
False

dict() == {}
True
```

Since dictionaries are mutable, they shouldn't be used as default values in Python functions, as that can lead to bugs. The usual way is to pass `None` as the default value and then check inside the function body if we got a dict sent in as an input. In this case, since `params` is passed straight to `self.http_session.get()`, you don't even need to check if it is `None` or not, since `requests` already does that.

```suggestion
            params: typing.Optional[typing.Dict] = None
```

Reply:
updated @ricardogsilva